src/validators/duration_distance.rs

```diff
@@ -38,65 +37,60 @@ fn max_speed(route_type: gtfs_structures::RouteType) -> f64 {
-                        IssueType::NullDuration,
-                        &*departure.stop,
-                    )
-                    .add_related_object(&*arrival.stop)
-                    .add_related_object(trip),
-                )
+                Some((Severity::Warning, IssueType::NullDuration))
             } else if duration > 0.0 && distance / duration > max_speed(route.route_type) {
-                result.push(
-                    Issue::new_with_obj(
-                        Severity::Information,
-                        IssueType::ExcessiveSpeed,
-                        &*departure.stop,
-                    )
-                    .add_related_object(&*arrival.stop)
-                    .add_related_object(trip),
-                )
+                Some((Severity::Information, IssueType::ExcessiveSpeed))
             } else if duration < 0.0 {
-                result.push(
-                    Issue::new_with_obj(
-                        Severity::Warning,
-                        IssueType::NegativeTravelTime,
-                        &*departure.stop,
-                    )
-                    .add_related_object(&*arrival.stop)
-                    .add_related_object(trip),
-                )
+                Some((Severity::Warning, IssueType::NegativeTravelTime))
             } else if distance / duration < 0.1 {
-                result.push(
-                    Issue::new_with_obj(Severity::Information, IssueType::Slow, &*departure.stop)
+                Some((Severity::Information, IssueType::Slow))
+            } else {
+                None
+            };
+
+            // we want to limit the number of duplicate, we we don't want an issue for all the trip between A&B
+            // we group all the issue by stops (and issue type)
+            if let Some((severity, issue_type)) = issue_kind {
+                // it's a bit of a trick, if we have an issue between A&B, we don't want a duplicate issue between B&A
+                let key = match departure.stop.id < arrival.stop.id {
+                    true => (
+                        departure.stop.id.clone(),
+                        arrival.stop.id.clone(),
+                        issue_type,
+                        severity,
+                    ),
+                    false => (
+                        arrival.stop.id.clone(),
+                        departure.stop.id.clone(),
+                        issue_type,
+                        severity,
+                    ),
+                };
+                let issue = issues_by_stops_and_type.entry(key).or_insert_with(|| {
+                    Issue::new_with_obj(severity, issue_type, &*departure.stop)
                         .add_related_object(&*arrival.stop)
-                        .add_related_object(trip),
-                )
+                });
+                issue.push_related_object(trip);
             }
         }
     }
 
-    Ok(result)
+    Ok(issues_by_stops_and_type
```

Comment:
what about iterating on `values()` ?

Reply:
I did not find a moving version of `.values()` (`.into_values` ?), but maybe it missed something